File: innerlibs1/htmlparser/CSSSelectors/PseudoClassSelectors/NthChildPseudoClass.vb

Imports HtmlParser
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace HtmlParser.PseudoClassSelectors
    <PseudoClassName("nth-child")> _
    Friend Class NthChildPseudoClass

        Inherits PseudoClass

        Protected Overrides Function CheckNode(node As HtmlElement, parameter As String) As Boolean

            Return node.Index = Integer.Parse(parameter) - 1
        End Function

    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================

File: InnerLibsCommon/HtmlParser/PseudoClassSelectors/NthChildPseudoClass.cs
namespace Extensions.Web.PseudoClassSelectors
{
    [PseudoClassName("nth-child")]
    internal class NthChildPseudoClass : PseudoClass
    {
        protected override bool CheckNode(HtmlNode node, string parameter) => node.Index == int.Parse(parameter) - 1;
    }
}